modules/caddyhttp/caddyhttp.go

```diff
@@ -135,15 +135,16 @@ func (app *App) Validate() error {
 	lnAddrs := make(map[string]string)
 	for srvName, srv := range app.Servers {
 		for _, addr := range srv.Listen {
-			netw, expanded, err := caddy.ParseNetworkAddress(addr)
+			listenAddr, err := caddy.ParseNetworkAddress(addr)
 			if err != nil {
 				return fmt.Errorf("invalid listener address '%s': %v", addr, err)
 			}
-			for _, a := range expanded {
-				if sn, ok := lnAddrs[netw+a]; ok {
-					return fmt.Errorf("server %s: listener address repeated: %s (already claimed by server '%s')", srvName, a, sn)
+			for i := listenAddr.FromPort; i <= listenAddr.ToPort; i++ {
+				addr := net.JoinHostPort(fmt.Sprintf("%s://%s", listenAddr.Network, listenAddr.Host), fmt.Sprintf("%d", i))
```

Comment:
There should be a method on the struct to do this for us.

(Also, this format is wrong -- these aren't URLs!)

Or, why not use `JoinNetworkAddress()`?